docs/metrics/README.md

```diff
@@ -417,6 +417,73 @@ SDK to reclaim unused metric points.
   Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#overflow-attribute)
   become stable, this feature will be turned on by default.
 
+#### Cardinality overflow example
+
+In the following example we set the maximum number of `MetricPoint`s allowed to
+be `3` for two metrics. This means that for these metrics, the SDK will only
+export measurements for up to `3` distinct key/value combinations.
+
+> [!NOTE]
+> One `MetricPoint` is reserved on every `Metric` for the special case where
+  there are no key/value pairs associated with a measurement. When choosing a
+  cardinality limit users should account for this special case.
```

Comment:
I understand this is the current implementation, do we want the user to even know this in the end? In other words, as a user, why would I need to understand this special case?

Reply:
I don't feel passionately about it. But it is interesting/surprising so it might be worth mentioning. In this example there is `AnotherFruitCounter`. It has a cardinality limit set to 3 but only 2 things are exported. Because of this reservation thing.

Reply:
> It has a cardinality limit set to 3 but only 2 things are exported. Because of this reservation thing.

@utpilla how much cost (e.g. perf) would it be if we want to hide this from the user? It'll be great if this can be made simpler.

Reply:
Here is my general suggestion:
1. We should try to make it simple, by *not* putting this burden on the user.
2. If we couldn't figure out a good way (e.g. perf suffers), take a different approach - keep the one-off but *hide* it from the user (@utpilla and I discussed about some ideas last week).
3. Worst case - we always ask the users to learn about and take care of something that we artificially introduced.

Reply:
I agree this reservation thing seems weird. IMO if we see a measurement with no tags we should just go and try to find a metric point to hold it as we do all other combinations. My guess would be its a quirk of the implementation. May or may not be able to solve it. But could we merge this PR and tackle that separately? We've always mentioned the reservation in the docs it wasn't something I added. If we are able to solve it then we can circle back and remove it?

Reply:
special casing for no tags and overflow tags is purely internal implementation details (and done for perf reasons.)

If user say cardinality limit is 1000, we currently reserve 2 spots for above, so users get only 998. We could instead pre-allocate 1002, so user gets his 1000, and we don't need to force user to learn about it. (in an advanced doc, we can state that overflow/no-tag case do not count towards this limit)

Reply:
> If user say cardinality limit is 1000, we currently reserve 2 spots for above, so users get only 998. We could instead pre-allocate 1002, so user gets his 1000, and we don't need to force user to learn about it. (in an advanced doc, we can state that overflow/no-tag case do not count towards this limit)

+1 👍 this is exactly one of the options @utpilla and I discussed last week.

Reply:
> We could instead pre-allocate 1002, so user gets his 1000, and we don't need to force user to learn about it. 

Agreed.

Reply:
> But could we merge this PR and tackle that separately?

@CodeBlanch Let's make the code changes first before adding the example? I think it would be simpler to just have the finalized example added instead of having another interim explanation of cardinality.

Reply:
Works for me I just closed this PR.